### ISSUE TITLE
Incompatibly "MonoAndroid,Version=v7.1 - Could not install package Xamarin.Android.Support.CustomTabs 26.0.2

### ISSUE THREAD

**Issue by bbl-Laobu** [OneSignal/OneSignal-Xamarin-SDK#81]
Just tried to install version 3.1.1 but I am getting the following error in the droid project during installation:

Could not install package 'Xamarin.Android.Support.CustomTabs 26.0.2'. You are trying to install this package into a project that targets 'MonoAndroid,Version=v7.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.

IOS and the shared PCL project had no issues. 

What to do next?

**Comment by jkasten2** [comment]
@bbl-Laobu The SDK version 3.1.1 currently requires your project to target Android 8.1 due to the way dependencies were set in it's NuGet dependencies. You will need to up this by right clicking on your Droid project and section Options. Then set this under General:
![image](https://user-images.githubusercontent.com/645861/38274831-bd595b3c-3744-11e8-8ccd-e36acd1fd8c9.png)

Just created PR #82 however to fix this so projects targeting 7.1 and 8.0 will work as well

**Comment by xalikoutis** [comment]
got this error too, just search in nuget and install it Xamarin.Android.Support.CustomTabs

**Comment by tidusjar** [comment]
When will 3.1.2 be available with this fix?

**Comment by jkasten2** [comment]
@tidusjar Yes, Target frameworks of Android 8.0 and Android 7.1 (default for new projects) will work after the next update is released.

Until then either my suggestion https://github.com/OneSignal/OneSignal-Xamarin-SDK/issues/81#issuecomment-378393272 or xalikoutis's https://github.com/OneSignal/OneSignal-Xamarin-SDK/issues/81#issuecomment-378393272 above will work in the mean time.

**Comment by tidusjar** [comment]
Thanks for the reply @jkasten2   I did set it to compile using 8.1 but I still got the same error. 

Do you know if there is a ETA on the next release?

**Comment by jkasten2** [comment]
@tidusjar I updated this ticket title to be more specific, the changes noted in PR #82 are only around MonoAndroid version compatibly for 7.1 and 8.0.

I replied to the other ticket https://github.com/OneSignal/OneSignal-Xamarin-SDK/issues/91#issuecomment-386207976 where this is an Incompatibly with another nuget package using an older version of `Xamarin.Android.Support.Compat`.

We are looking at shipping a release to nuget next week.

**Comment by taublast** [comment]
What what that breaking change introduced after 3.0.1 wich was all compatible:
Xamarin.GooglePlayServices.Gcm (>= 27.0.0)
Xamarin.Android.Support.v4 (>= 23.0.1.3)

and then every version is 
Xamarin.Android.Whatever (>= YY.0.0 && < XX.X.0)

Today again all these versions are not compatible with the latest xamarin, that is already at 28 and we are restricted to <27.2.0. Why are these restrictions in place, pardon me but it's a very bad practice.

Just forced to use 3.0.1 as xamarin updates often and onesignal just blocks updating xamarin on every xamarin release.

**Comment by bbl-Laobu** [comment]
For me, Xamarin.Esentials is now requesting minimum Xamarin.Android.Support.Compat (>= 28.0.0.1). So, an update for the OneSignal package would be great. Any ETA available?

**Comment by JoseluDAM2000** [comment]
I have managed to install both adding the dependencies that gave conflicts. 

In this case i use the NuGet Package Manager to install manually the Xamarin.Android.Support.CustomTabs 27.0.2
```
 joseluApp.Android -> Xamarin.Essentials 1.1.0 -> Xamarin.Android.Support.CustomTabs (>= 28.0.0.1) 
 joseluApp.Android -> Com.OneSignal 3.3.0-beta -> Xamarin.Android.Support.CustomTabs (>= 27.0.0 && < 27.2.0).
```
I do that few times, installing the Core.Utils, Support.Compact and that's it, the app looks to work without issues.

I think it's a acceptable solution until the problem is solved.

**Comment by jkasten2** [comment]
@JoseluDAM2000 @bbl-Laobu We did remove the upper bounds OneSignal's dependencies in the final [3.3.0 release](https://github.com/OneSignal/OneSignal-Xamarin-SDK/releases/tag/3.3.0). Updating to this should solve this issue. Let us know if it does not and we can reopen this issue.
Thanks.

**Comment by omrivm** [comment]
I'm using 3.3.0 and i still get this error

**Comment by omrivm** [comment]
This is a blocker for us, since we can't continue the development.
Any ETA on this issue?

**Comment by JoseluDAM2000** [comment]
As i wrote to you into the #148:
I managed to avoid that failure by install que necesary android libraries by hand:
`"Xamarin.Android.Support.Compat" Version 27.0.2` 
`"Xamarin.Android.Support.Core.Utils" Version 27.0.2`

**Comment by omrivm** [comment]
Solved it (see #148 )
